src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -526,4 +530,46 @@ object AppDefinition {
       }
     }
   }
+
+  def residentUpdateIsValid(from: AppDefinition): Validator[AppDefinition] = {
+    def changeNoVolumes: Validator[AppDefinition] = new Validator[AppDefinition] {
+      override def apply(to: AppDefinition): Result = {
+        val fromVolumes = from.persistentVolumes
+        val toVolumes = to.persistentVolumes
+        def sameSize = fromVolumes.size == toVolumes.size
+        def noChange = from.persistentVolumes.forall { fromVolume =>
+          toVolumes.find(_.containerPath == fromVolume.containerPath).contains(fromVolume)
```

Comment:
The `contains` method will use equals, wich does a deep comparison of the volumes.
If the size (or any parameter) changes, it will fail.